### PR TITLE
add timeout for upsteam fetch

### PR DESCRIPTION
**Pull request by ZhongFuze**
add timeout for upsteam fetch

- hyper https client: add utils method ``request_with_timeout``
- gqlclient: use `tokio::time::timeout` in request logic

### CODE REVIEW

File: src/upstream/dotbit/tests.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ async fn test_smoke_dotbit_by_dotbit_identity() -> Result<(), Error> {
     let found = Identity::find_by_platform_identity(&db, &target.platform()?, &target.identity()?)
         .await?
         .expect("Record not found");
-
+    print!("found {:?}", found);
```
Comment (nykma) [inline src/upstream/dotbit/tests.rs:18]: Change this to `debug!`

File: src/util/mod.rs [hunk 2]
```diff
@@ -33,10 +35,32 @@ pub fn timestamp_to_naive(ts: i64, ms: u32) -> NaiveDateTime {
...
 pub fn make_client() -> Client<HttpsConnector<HttpConnector>> {
     let https = HttpsConnector::new();
+    // let mut http = HttpConnector::new();
+    // http.set_connect_timeout(Some(std::time::Duration::from_secs(5)));
+    // let https = HttpsConnector::new_with_connector(http);
 
     Client::builder().build::<_, hyper::Body>(https)
 }
 
+pub async fn request_with_timeout(
```
Comment (fengshanshan) [inline src/util/mod.rs:45]: Shall we put timeout as one of the input params for different upstreams?

**Review by fengshanshan — DISMISSED** [review]

**Review by nykma — CHANGES_REQUESTED** [review]
Small tweaks needed

File: src/util/mod.rs (hunk 2 above)
Comment (nykma) [reply src/util/mod.rs:45]: Agree, but good for now to observe if issue is mitigated or not.

File: src/upstream/dotbit/tests.rs (hunk 1 above)
Comment (ZhongFuze) [reply src/upstream/dotbit/tests.rs]: fixed

**Review by nykma — APPROVED** [review]

**Pull request merged**